### PR TITLE
CMakeLists.txt:put binaries in correct locations when installing

### LINKED ISSUE #1710: Binaries installed in odd directory under linux
Revision 7224.1.5 removed WL_INSTALL_BINDIR, instead opting to use CMAKE_INSTALL_PREFIX. However, binary files by default are now installed into CMAKE_INSTALL_PREFIX directly. I believe this is expected for Windows and MacOS, but under Linux the expectation is that binary files go into CMAKE_INSTALL_PREFIX/bin. In particular, a relatively default install prefix is /usr, but would result in the widelands executable being placed as "/usr/widelands". One option might be something like

IF(ARGS_WIN32)
install(TARGETS ${NAME} DESTINATION "." COMPONENT ExecutableFiles)
ELSE(ARGS_WIN32)
install(TARGETS ${NAME} DESTINATION "bin/" COMPONENT ExecutableFiles)
ENDIF(ARGS_WIN32)

in cmake/WlFunctions.cmake. There are potentially other ways around this, another one I know is to have a "BINDIR" type variable that is set in some OS-specific setup function, but that's basically just bringing back WL_INSTALL_BINDIR, and I assume this was removed for some specific reason.

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2018-08-09T22:40:29Z
 * owner: uilliam
 * the launchpad url was https://bugs.launchpad.net/bugs/1786353

### PR DESCRIPTION
**Pull request by andy5995**
(closes #1710)

### CODE REVIEW

**Review by matthiakl — COMMENTED** [review]
Please check xdg/CMakeLists.txt and replace the relative `..` paths there with the now correct absolute CMAKE_INSTALL_PREFIX.

**Comment by andy5995** [comment]
> Please check xdg/CMakeLists.txt and replace the relative `..` paths there with the now correct absolute CMAKE_INSTALL_PREFIX.

What about here? https://github.com/widelands/widelands/blob/0621ce2fe1a3825ae6669a424a700de5312fb84f/xdg/CMakeLists.txt#L21 ?

Does the `..` just need to be removed?

**Comment by matthiakl** [comment]
> Does the .. just need to be removed?

Yes, this results from the strange expectation, that CMAKE_INSTALL_PREFIX points to /usr/bin instead of /usr

File: CMakeLists.txt [hunk 1]
```diff
@@ -554,6 +554,13 @@ install(
   PATTERN "CMakeLists.txt" EXCLUDE
 )
 
+# To override this, use '-DWL_INSTALL_BINDIR=<absolute path>' when configuring or change
+# 'CMAKE_INSTALL_PREFIX'.
+# Example: to have the bin installed to "/usr/games", just use '-DCMAKE_INSTALL_PREFIX=/usr'
+# and don't specify '-DWL_INSTALL_BINDIR' when configuring.
+# Note: 'CMAKE_INSTALL_PREFIX' defaults to /usr/local on UNIX and c:/Program Files/${PROJECT_NAME} on Windows
+set(WL_INSTALL_BINDIR "${CMAKE_INSTALL_PREFIX}/games" CACHE PATH "Destination of executable files (if installing)")
```
Comment (andy5995) [inline CMakeLists.txt:563]: Does anyone think we need a condition  here to specify a different default directory if installing on Windows? If I read the cmake docs right, the binaries would get installed to `c:/Program Files/widelands/games/`

**Review by matthiakl — APPROVED** [review]
Looks good now.
@Noordfrees, we need to inform the package maintainers about this change, so they can update their paths configuration.

File: CMakeLists.txt (hunk 1 above)
Comment (matthiakl) [reply CMakeLists.txt:563]: Windows builds do not get installed by `make` but instead use the innosetup, mac os also has a separate install method. This should be unaffected by this change. However to be sure I will test the Windows installer and maybe @klaus-halfmann can check on MacOS (go to "Checks" -> "Artifacts")

Edit: Windows installer is still working as expected

**Comment by Noordfrees** [comment]
The daily PPA is failing now:
https://launchpadlibrarian.net/599328127/buildlog_ubuntu-impish-amd64.widelands_2%3A1.1~bzr10257-202205011033~ubuntu21.10.1_BUILDING.txt.gz
How do the paths in `debian/*` need to be adapted?

File: CMakeLists.txt [hunk 2]
```diff
@@ -555,6 +555,13 @@ install(
...
```
Comment (matthiakl) [inline CMakeLists.txt:563]: > The daily PPA is failing now

Uh, oh. The value for WL_INSTALL_BINDIR here is set after including the src directory, so it is not picked up properly if not specified on the command line.

**Pull request merged**